discovery/service_mock_test.go

```diff
@@ -0,0 +1,117 @@
+package discovery
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type ConsulClientWrapperMock struct {
+	mock.Mock
+}
+
+// Health function mock
+func (hw *ConsulClientWrapperMock) Health() IHealth {
+	args := hw.Called()
+	if health, ok := args.Get(0).(IHealth); ok {
+		return health
+	}
+	return nil
+}
+
+type HealthWrapperMock struct {
+	mock.Mock
+}
+
+// Service function mock
+func (hw *HealthWrapperMock) Service(service, tag string, passingOnly bool, q *api.QueryOptions) ([]*api.ServiceEntry, *api.QueryMeta, error) {
+	args := hw.Called()
+	return args.Get(0).([]*api.ServiceEntry), args.Get(1).(*api.QueryMeta), args.Error(2)
+}
+
+func TestShouldGetEndpointWhenOneInstanceExists(t *testing.T) {
+	host := "localhost"
+	port := 12345
+	expectedHost := fmt.Sprintf("%s:%d", host, port)
+
+	consulClientMock := prepareConsulClientMock(host, port, 1)
+	services := NewServices(consulClientMock, 1)
+
+	instance, err := services.GetEndpoint("my-service")
+
+	require.NoError(t, err)
+	require.Equal(t, instance.Host, expectedHost)
+}
+
+func TestShouldGetEndpointWhenTwoInstancesExists(t *testing.T) {
+	host := "localhost"
+	port := 12345
+	expectedHost1 := fmt.Sprintf("%s:%d", host, port)
+	expectedHost2 := fmt.Sprintf("%s:%d", host, port+1)
+
+	consulClientMock := prepareConsulClientMock(host, port, 2)
+	services := NewServices(consulClientMock, 5)
+
+	instance, err := services.GetEndpoint("my-service")
+
+	require.NoError(t, err)
+	require.True(t, instance.Host == expectedHost1 || instance.Host == expectedHost2)
+}
+
+func TestShouldGetEndpointReturnEndpointAfterTwiceCalling(t *testing.T) {
```

Comment:
If this tests checks double calling then what is the sleep inside for?

Reply:
simulating delay for next cache revalidation